### PR TITLE
Feature/product details

### PR DESCRIPTION
**Pull request by sharansp**
Testing build check

### CODE REVIEW

File: src/app/product-details/product-details.component.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+				} = respData
+				if (hits && hits.hit && hits.hit[0].id) {
+					this.updateProductDetails(hits)
+				} else {
+					let query = {
+						q: `(term field=_id ${this.itemId})`,
+						size: 10,
+						qdotparser: 'structured'
+					}
+					this._productsService.get(query).
```
Comment (reddy-s) [inline src/app/product-details/product-details.component.ts]: @sharansp , why are we doing two calls here?

File: src/app/product-details/product-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			this.itemId = params.id;
+		});
+
+		// get product details
+		if (this.itemId) {
+			this._searchItemService.responsePoductListState.subscribe(respData => {
+				let {
+					hits
+				} = respData
+				if (hits && hits.hit && hits.hit[0].id) {
```
Comment (reddy-s) [inline src/app/product-details/product-details.component.ts]: Also, why do we need to check for so many things?

Comment (sharansp) [reply src/app/product-details/product-details.component.ts]: I have updated this, checking only for id now

File: src/app/product-details/product-details.component.ts (hunk 1 above)
Comment (sharansp) [reply src/app/product-details/product-details.component.ts]: @reddy-s 
1. this._productsService.get(query) -> 
- is when product with specific id link is opened.
- we make fresh call

2. this._searchItemService.responsePoductListState.subscribe(respData => {
- Is to check if already data is already present for specific id

**Pull request merged**